### DISCUSSION TITLE
Custom plug-in running error

### DISCUSSION THREAD

**Discussion by caiwenxuan** [golangci/golangci-lint#5237] [Q&A] [answered]
### Welcome

- [X] Yes, I'm using a binary release within 2 latest [releases](https://github.com/golangci/golangci-lint/releases). Only such installations are supported.
- [X] Yes, I've searched similar [issues on GitHub](https://github.com/golangci/golangci-lint/issues) and didn't find any.
- [X] Yes, I've read the `typecheck` section of the [FAQ](https://golangci-lint.run/welcome/faq/#why-do-you-have-typecheck-errors).
- [X] Yes, I've tried with the standalone [linter](https://golangci-lint.run/usage/linters/) if available (e.g., gocritic, go vet, etc.).
- [X] I agree to follow this project's [Code of Conduct](https://github.com/golangci/golangci-lint?tab=coc-ov-file)

### Description of the problem

I used golangci-lint with tag=v1.62.2, and confirmed that the go.mod version used to generate the custom plugin is the same as tag=v1.62.2, and the go version is the same. Why is it reporting an error?

![image](https://github.com/user-attachments/assets/d3f1cfd7-3cde-4797-b455-31a7979bf0e2)
The package version corresponds to the link below
https://github.com/golangci/golangci-lint/blob/89476e7a1eaa0a8a06c17343af960a5fd9e7edb7/go.mod

### Version of golangci-lint

<details>

```console
$ golangci-lint --version
# Paste output here
root@683283a55bb1:/work/test/example-plugin-linter# golangci-lint version
golangci-lint has version 1.62.2 built with go1.23.3 from 89476e7a on 2024-11-25T14:16:01Z
```

</details>

### Configuration

<details>

```console
# paste configuration file or CLI flags here
```

</details>

### Go environment

<details>

```console
$ go version && go env
# paste output here
go version go1.23.3 linux/amd64
GO111MODULE=''
GOARCH='amd64'
GOBIN=''
GOCACHE='/root/.cache/go-build'
GOENV='/root/.config/go/env'
GOEXE=''
GOEXPERIMENT=''
GOFLAGS=''
GOHOSTARCH='amd64'
GOHOSTOS='linux'
GOINSECURE=''
GOMODCACHE='/go/pkg/mod'
GONOPROXY=''
GONOSUMDB=''
GOOS='linux'
GOPATH='/go'
GOPRIVATE=''
GOPROXY='https://proxy.golang.org,direct'
GOROOT='/usr/local/go'
GOSUMDB='sum.golang.org'
GOTMPDIR=''
GOTOOLCHAIN='local'
GOTOOLDIR='/usr/local/go/pkg/tool/linux_amd64'
GOVCS=''
GOVERSION='go1.23.3'
GODEBUG=''
GOTELEMETRY='local'
GOTELEMETRYDIR='/root/.config/go/telemetry'
GCCGO='gccgo'
GOAMD64='v1'
AR='ar'
CC='gcc'
CXX='g++'
CGO_ENABLED='1'
GOMOD='/work/test/example-plugin-linter/go.mod'
GOWORK=''
CGO_CFLAGS='-O2 -g'
CGO_CPPFLAGS=''
CGO_CXXFLAGS='-O2 -g'
CGO_FFLAGS='-O2 -g'
CGO_LDFLAGS='-O2 -g'
PKG_CONFIG='pkg-config'
GOGCCFLAGS='-fPIC -m64 -pthread -Wl,--no-gc-sections -fmessage-length=0 -ffile-prefix-map=/tmp/go-build1315462551=/tmp/go-build -gno-record-gcc-switches'
```

</details>

### Verbose output of running

<details>

```console
$ golangci-lint cache clean
$ golangci-lint run -v
# paste output here
root@683283a55bb1:/work/test/example-plugin-linter# golangci-lint cache clean
root@683283a55bb1:/work/test/example-plugin-linter# golangci-lint run -v
INFO golangci-lint has version 1.62.2 built with go1.23.3 from 89476e7a on 2024-11-25T14:16:01Z
INFO [config_reader] Config search paths: [./ /work/test/example-plugin-linter /work/test /work / /root]
INFO [config_reader] Used config file .golangci.yml
Error: build linters: unable to load custom analyzer "example": example.so, plugin: not implemented
Failed executing command with error: build linters: unable to load custom analyzer "example": example.so, plugin: not implemented
```

</details>

### A minimal reproducible example or link to a public repository

<details>

```go
// add your code here
```

</details>

### Validation

- [X] Yes, I've included all information above (version, config, etc.).

### Supporter

- [X] I am a sponsor/backer through [GitHub](https://github.com/sponsors/golangci/) or [OpenCollective](https://opencollective.com/golangci-lint)

**Comment by ldez** [comment]
Sorry, but I can see 0 error in the screenshot.

The text inside the `go.mod` is a comment coming from https://github.com/golangci/example-plugin-linter/blob/5b227aea582953badf9ebb040a5fbbc36f265b6f/go.mod#L3

Maybe your screenshot is incomplete, so copy-paste the text (please no screenshot).

**Comment by ldez** [comment]
From your logs:

```
Error: build linters: unable to load custom analyzer "example": example.so, plugin: not implemented
Failed executing command with error: build linters: unable to load custom analyzer "example": example.so, plugin: not implemented
```

Your problem is that your plugin doesn't implement the right elements.

**Reply by ldez** [reply, marked as answer]
You can read the documentation: https://golangci-lint.run/plugins/go-plugins/

You can also use the new plugin system https://golangci-lint.run/plugins/module-plugins/

**Comment by caiwenxuan** [comment]
golangci-lint is a binary file downloaded using https://github.com/golangci/golangci-lint/releases/tag/v1.62.2 curl
`./golangci-lint` is compiled.
```
git clone -b v1.62.2 https://github.com/golangci/golangci-lint.git
make build
```
From the error message, is it because the CGO_ENABLED=0 of the binary file downloaded by curl caused golangci-lint run to fail, while the binary file I generated myself succeeded with CGO_ENABLED=1

> https://github.com/golangci/golangci-lint/issues/1276#issuecomment-665903858

**Reply by ldez** [reply]
This is explained inside the documentation: https://golangci-lint.run/plugins/go-plugins/

Please read it.
